common/rpc/encryption/localStoreCertProvider.go

```diff
@@ -77,57 +92,142 @@ func (s *localStoreCertProvider) FetchServerCertificate() (*tls.Certificate, err
 	return s.serverCert, nil
```

Comment:
`s.serverCert` would never get set in this code path, right? I think this line should be in `FetchServerCertificate()` where the caching logic is, and `fetchServerCertificateFromFile()` and  `fetchServerCertificateFromInline()` shouldn't touch it. Maybe go even further, and pass arguments to them explicitly instead of `(s *localStoreCertProvider)`, so that they son't have access to `s`?